HeterogeneousCore/AlpakaTest/plugins/alpaka/TestAlgo.dev.cc

```diff
@@ -0,0 +1,38 @@
+// Check that ALPAKA_HOST_ONLY is not defined during device compilation:
+#ifdef ALPAKA_HOST_ONLY
+#error ALPAKA_HOST_ONLY defined in device compilation
+#endif
+
+#include <alpaka/alpaka.hpp>
+
+#include "DataFormats/PortableTestObjects/interface/alpaka/TestDeviceCollection.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+#include "TestAlgo.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class TestAlgoKernel {
+  public:
+    template <typename TAcc>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, portabletest::TestDeviceCollection::View view, int32_t size) const {
+      int32_t idx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
```

Comment:
How about generalizing this kernel to work on arbitrary grid parameters? (for the sake of giving an example)

Reply:
Sure... done for an arbitrary number of blocks and threads, still assuming a single element per thread.

A more meaningful example should use a varying number of elements per threads - but for that I would prefer to bring in the helper functions we have in pixeltrack-standalone (future PR).

Reply:
> A more meaningful example should use a varying number of elements per threads - but for that I would prefer to bring in the helper functions we have in pixeltrack-standalone (future PR).

Good point, I agree the helper functions would be very beneficial for a meaningful example.